clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -840,22 +840,14 @@ public void commit(final Map<TopicPartition, Long> offsets, CommitType commitTyp
      * every rebalance and also on startup. As such, if you need to store offsets in anything other than Kafka, this API
      * should not be used.
      * <p>
-     * Asynchronous commits (i.e. {@link CommitType#ASYNC} will not block. Any errors encountered during an asynchronous
-     * commit are either passed to the callback (if provided) or silently discarded. Synchronous commits (i.e.
-     * {@link CommitType#SYNC}) block until either the commit succeeds or an unrecoverable error is encountered. In
-     * this case, the error is either passed to the callback (if provided) or thrown to the caller.
+     * This is an asynchronous call and will not block. Any errors encountered are either passed to the callback
+     * (if provided) or discarded.
      *
-     * @param commitType Whether or not the commit should block until it is acknowledged.
      * @param callback Callback to invoke when the commit completes
      */
     @Override
-    public void commit(CommitType commitType, ConsumerCommitCallback callback) {
-        acquire();
-        try {
-            commit(subscriptions.allConsumed(), commitType, callback);
-        } finally {
-            release();
-        }
+    public void commit(OffsetCommitCallback callback) {
+        commit(subscriptions.allConsumed(), callback);
```

Comment:
Shouldn't this still have the `acquire`/`release` calls since it invokes `subscriptions.allConsumed()` before calling the other form of `commit`?